dashboard/tests/test_dashboard.py

```diff
@@ -925,11 +925,12 @@ def test_dashboard_does_not_depend_on_serve():
     # Check that Serve-dependent features fail
     try:
         response = requests.get(f"http://{agent_url}/api/serve/deployments/")
-        assert response.status_code == 501
     except Exception as e:
         # Fail to connect to service is fine.
         print(e)
-        assert True
+    else:
```

Comment:
wow, I did not know that you could use an `else` block after an exception handler until today... this is tricky and seems like it might confuse people in the future. can you rewrite this in a way to avoid this syntax?

in general we shouldn't be catching bare `Exception` types because it causes subtle bugs like the one you uncovered here. maybe catch a more specific error in that block and then keep the assert in the `try` block?

Reply:
Switched to catching `requests.ConnectionError`!